docs/notebooks/visualizing_with_tensorboard.pct.py

```diff
@@ -0,0 +1,124 @@
+
+# %%
+# Clear any logs from previous runs
+# !rm -rf logs/tensorboard
+
+summary_writer = tf.summary.create_file_writer("logs/tensorboard")
+trieste.logging.set_tensorboard_writer(summary_writer)
+
+# %% [markdown]
+# We can now also load the TensorBoard extension, though at this point there will not be any data to dispay.
+
+# %% magic_args="echo Loading TensorBoard..." language="script"
+#
+# %load_ext tensorboard
+# %tensorboard --logdir "logs/tensorboard"
+
+# %% [markdown]
+# ![TensorBoard empty page](figures/tensorboard_empty.png)
+
+# %% [markdown]
+# ## Running and tracking the Bayesian Optimizer
+#
+# By setting the summary writer, we tell trieste to log relevant information during optimization. While the optimization is running, we can refresh TensorBoard to see its progress.
+
+# %%
+num_steps = 15
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+result, history = bo.optimize(num_steps, initial_data, model).astuple()
+
+
+# %% [markdown]
+# The data remains available when the optimization is over.
+#
+# ![TensorBoard graphs](figures/tensorboard_bo.png)
+
+# %% [markdown]
+# ## Logging additional model parameters
+#
+# When logging is enabled, trieste decides what information is interesting enough to log. This includes objective and acquisition function values, and some (but not all) model parameters. To log additional model parameters, you can define your own model subclass and override the `log` method. For example, the following GPR subclass also logs the average lengthscale at each step.
+
+# %%
+class GPRExtraLogging(trieste.models.gpflow.GaussianProcessRegression):
+
+    def log(self, context):
+        """
+        Log model-specific information at a given optimization step.
+
+        :param context: A context string to use when logging.
+        """
+        super().log(context)
+        summary_writer = trieste.logging.get_tensorboard_writer()
+        if summary_writer:
+            with summary_writer.as_default(step=trieste.logging.get_step_number()):
+                tf.summary.scalar(
+                    f"{context}.kernel.lengthscales.mean",
+                    np.mean(self.get_kernel().lengthscales)
+                )
+
+
+model = GPRExtraLogging(gpr)
+
+# %% [markdown]
+# Running with this model now also produces logs for the mean lengthscale.
+
+# %%
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+result, history = bo.optimize(num_steps, initial_data, model).astuple()
+
+# %% [markdown]
+# ![TensorBoard custom graphs](figures/tensorboard_custom.png)
+
+# %% [markdown]
+# ## Using Tensorboard with Ask-Tell Optimization
+#
+# To use Tensorboard logging with the [Ask-Tell interface](ask_tell_optimization.ipynb), you must also explicitly set the optimization step number before calling ask or tell:
+
+# %%
+for step in range(num_steps):
+    trieste.logging.set_step_number(step)
+    # new_point = ask_tell.ask()
```

Comment:
this is still commented out? you need to import asktell as well...

Reply:
this is just an example rather than full running code (I'll add a comment to make that clear)

Reply:
actually, i'll turn it into a working example (and use a new summary writer too)